macroslib/src/cpp/rust_str.h

```diff
@@ -16,7 +16,8 @@ struct RustStrView {
     bool empty() const noexcept { return this->len == 0; }
     std::string to_std_string() const { return std::string{ data, len }; }
 #endif
-#if __cplusplus > 201402L
+#if __cplusplus > 201402L && (!defined(__GNUC__) || defined(__clang__))
+// GCC5 reports `__cplusplus > 201402L`, but does not support `std::string_view`.
```

Comment:
Looks strange, Why code and comment do not match?

According to comment the code should be

```
#if __cplusplus > 201402L &&  (!defined(__GNUC__) || __GNUC__ > 5)
```

but instead of `__GNUC__ > 5` it check `__clang__` for some reason?

And looks like this completely unrelated issue, may be it is possible to move this change into separate commit?

Reply:
Clang also defines `__GNUC__`, so, if you need code specific for GCC, but not for clang, you need to test `defined(__GNUC__) && !defined(__clang__)`.
The condition is reversed here, becouse the code should be included if we are NOT in GCC.

I'll rebase it into separate commits.

Reply:
> Clang also defines __GNUC__

Ok, but what about gcc with version 6-8?

should it be:
```
#if __cplusplus > 201402L &&  (!defined(__GNUC__) || defined(__clang__) || __GNUC__ > 5)
```

Reply:
@Dushistov 
I've added more comprehensive test if `string_view` is supported and splited PR into two commits.